api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -477,27 +482,27 @@ public void setAutoExpireDateBasedOnDuration() {
 			setAutoExpireDate(getDosingInstructionsInstance().getAutoExpireDate(this));
 		}
 	}
-	
+
 	public String toString() {
 		String prefix = DISCONTINUE == getAction() ? "DC " : "";
 		return prefix + "DrugOrder(" + getDose() + getDoseUnits() + " of "
-		        + (getDrug() != null ? getDrug().getName() : "[no drug]") + " from " + getDateActivated() + " to "
+		        + (getDrugNonCoded() != null ? getDrugNonCoded() : (getDrug() != null ? getDrug().getName() : "[no drug]")) + " from " + getDateActivated() + " to "
```

Comment:
I agree it is not good to call the service layer from domain objects but sometimes we have to e.g we already do this for Obs for some good reasons. Since we updated the design of this and said DrugOrder.drug and DrugOrder.drugNonCoded should be mutually exclusive then i guess you can ignore my comment